core/primitives/src/test_utils.rs

```diff
@@ -306,6 +306,95 @@ impl BlockHeader {
     }
 }
 
+/// Builder class for blocks to make testing easier.
+/// Example usage:
+///
+/// let test_block = TestBlockBuilder::new(prev, signer).height(33).build();
+pub struct TestBlockBuilder {
+    prev: Block,
+    signer: Arc<dyn ValidatorSigner>,
```

Comment:
Given that most builders will be immediately consumed to construct a block, it would seem to me that `&dyn ValidatorSigner` would be a pretty painless option here, if only to make the use-location a little more convenient (where you'd be able to just pass in a `&signer` rather than having to wrap it into an `Arc` and then clone it.)

But this is fine either way.

Reply:
ack.